### ISSUE TITLE
Spawning the motor boat causes the game to hang up

### ISSUE THREAD

**Issue by Coolthulhu** [cataclysmbnteam/Cataclysm-BN#2377]
### Describe the bug

Attempting to spawn the motor boat will cause the game to lock up for at least a few minutes. Most likely forever.

### Steps To Reproduce

1. Debug-spawn the motor boat

### Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 22.04.2 LTS; Release: 22.04; Codename: jammy; 
- Game Version: 01dd1851729 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [bn],
    Magiclysm [magiclysm],
    Manual Bionic Installation [manualbionicinstall],
    Elevated bridges [elevated_bridges],
    TESTING DATA [test_data]
]

### Additional context

Found while testing #2339

**Comment by chaosvolt** [comment]
```
 DEBUG    : map::board_vehicle: vehicle not found

 FUNCTION : board_vehicle
 FILE     : D:\a\Cataclysm-BN\Cataclysm-BN\src\map.cpp
 LINE     : 1060
 VERSION  : BN 36191cb
```

This happens when [0, 0] on the vehicle (or whichever one tends to be marked as the "core" of the vehicle, the first tile that'll be highlighted on the vehicle screen) is one the player can't normally access, most often because it's blocked by a quarterpanel or board. It seems that bare frames with no aisles or seats on it also don't count as boardable in this case, for the purposes of what that debug message is looking for.

In the case of the motor boat, [0, 0] seems to be the back left corner of it, which has only a frame plus a boat hull:
![image](https://user-images.githubusercontent.com/11582235/224743648-482032ca-129d-43bc-b3c3-6769e356cf79.png)

**Comment by olanti-p** [comment]
Yeah, many debug-spawnable vehicles have this issue. Portable Generator, for example.

**Comment by chaosvolt** [comment]
The single-tile utility vehicles probably can't be fixed depending on what this error actually checks for (if it's nothing actually vital then removing it outright or making it also accept `MOUNTABLE` parts are options, but I assume this is used for something useful and not just debug spawning), but for boats it can be fixed the by altering its vehiclepart definitions so it starts with the part with the seat, like most vehicles.

**Comment by olanti-p** [comment]
As this was caused by backtrace performance, it's hopefully fixed by #3101
